zrcs/zrc-3.md

```diff
@@ -0,0 +1,431 @@
+
+### A. Roles
+
+| Name | Description
+|--|--|
+| Contract Owner | The owner of the contract initialized by the creator of the contract. |
+| Token Owner | A user (identified by her address) that owns a token.  |
+| Teller | A signing authority identified by its `_sender` address within a contract call that can only be executed if the teller the Teller has included a valid signed metatransaction specifying that transaction parameters and the `Token Owner`'s signature. A `Token Owner` can now trust that no authority can move funds without the `Token Owner`'s expressed consent. |
+
+### B. Error Codes
+
+The contract must define the following constants for use as error codes for the `Error` event.
+
+| Name | Type | Code | Description
+|--|--|--|--|
+| `CodeNoAuthorized` | `Int32` | `-1` | Emit when the transition call is unauthorized for a given user.
+| `CodeNotFound` | `Int32` | `-2` | Emit when a value is missing.
+| `CodeBadRequest` | `Int32` | `-3` | Emit when the transition call is somehow incorrect.
+| `CodeTokenExists`| `Int32` | `-4` | Emit when trying to create a token that already exists.
+| `CodeNonceError` | `Int32` | `-5` | Emit when the transition call tries to use a nonce lower than the accounts present nonce.
+| `CodeUnexpectedError` | `Int32` | `-6` | Emit when the transition call runs into an unexpected error.
+
+
+### C. Immutable Variables
+
+| Name                 | Type          | Description                                                           |
+| -------------------- | ------------- | --------------------------------------------------------------------- |
+| `contractOwner`      | `ByStr20`     | The owner of the contract initialized by the creator of the contract. |
+| `name`               | `String`      | The name of the fungible token.                                       |
+| `symbol`             | `String`      | The symbol of the fungible token.                                     |
+| `default_tellers`  | `List ByStr20`| The adddresses set as default for tellers all token holders.        |
+| `decimals`           | `Uint32`      | The number of decimal places a token can be divided by.               |
+
+### D. Mutable Fields
+
+| Name                | Type                                                              | Description                                                                                                                                                           |
+| ------------------- | ----------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `total_tokens` | `Uint128 = Uint128 0` |    Total amount of tokens.       |
+| `revokedDefaultTeller` | `Map ByStr20 (Map ByStr20 Bool) = Emp ByStr20 (Map ByStr20 Bool)` |    Mapping of `default_tellers` that have been revoked by token hodlers.       |
```

Comment:
token hodlers -> token holders (or is that intentiona?)

Reply:
It wasn't, good catch though. I've seen the word so many times now that either spellings can pass my mental spellchecker.